src/components/Button.tsx

```diff
@@ -53,8 +57,17 @@ export function Button(props: ButtonProps) {
   const { buttonProps, isPressed } = useButton(
     {
       ...ariaProps,
-      onPress: typeof onPress === "string" ? noop : onPress,
-      elementType: typeof onPress === "string" ? "a" : "button",
+      onPress: asLink
+        ? noop
+        : (e) => {
+            const result = onPress(e);
+            if (isPromise(result)) {
+              setIsDisabled(true);
+              result.finally(() => setIsDisabled(false));
```

Comment:
Praise: Nice! I forgot about `.finally`

Reply:
@bdow I'm reading this late, but I wonder if we need to be careful about "stomping" the user's `disabled` prop...

Like maybe after the call is done, the user starts passing in `disabled=false`, but since our `setIsDisabled` runs after the `useEffect` (I think it might be a race condition about if it does/does not), then we might overwrite it...

So maybe this `setIsDisabled true/false` should be a dedicated `useState` like "setInFlight true/false`, and then we treat disabled as an `or` of the prop or in flight?

Reply:
lol @stephenh... I was going to make that _exact_ comment then decided to just work on it myself.

Reply:
Reverting this change, will bring back and update this in another PR